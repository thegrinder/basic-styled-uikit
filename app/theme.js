const theme = {
  uiKit: {
    buttons: {
      btnTypes: {
        default: {
          regular: {
            color: '#222',
            bgColor: 'transparent',
            borderColor: '#e5e5e5',
          },
          hover: {
            color: '#222',
            bgColor: 'transparent',
            borderColor: '#b2b2b2',
          },
          pressed: {
            color: '#222',
            bgColor: 'transparent',
            borderColor: '#999',
          },
        },
        primary: {
          regular: {
            color: '#fff',
            bgColor: '#1e87f0',
            borderColor: 'transparent',
          },
          hover: {
            color: '#fff',
            bgColor: '#0f7ae5',
            borderColor: 'transparent',
          },
          pressed: {
            color: '#fff',
            bgColor: '#0e6dcd',
            borderColor: 'transparent',
          },
        },
        secondary: {
          regular: {
            color: '#fff',
            bgColor: '#222',
            borderColor: 'transparent',
          },
          hover: {
            color: '#fff',
            bgColor: '#151515',
            borderColor: 'transparent',
          },
          pressed: {
            color: '#fff',
            bgColor: '#080808',
            borderColor: 'transparent',
          },
        },
        danger: {
          regular: {
            color: '#fff',
            bgColor: '#f0506e',
            borderColor: 'transparent',
          },
          hover: {
            color: '#fff',
            bgColor: '#ee395b',
            borderColor: 'transparent',
          },
          pressed: {
            color: '#fff',
            bgColor: '#ec2147',
            borderColor: 'transparent',
          },
        },
      },
      btnSizes: {
        small: {
          padding: '0 15px',
          lineHeight: '28px',
          fontSize: '14px',
        },
        default: {
          padding: '0 30px',
          fontSize: '14px',
          lineHeight: '38px',
        },
        large: {
          padding: '0 40px',
          lineHeight: '53px',
          fontSize: '16px',
        },
      },
    },
  },
};

export default theme;
