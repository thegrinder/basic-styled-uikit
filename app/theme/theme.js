import buttonTheme from './button-theme';
import linkTheme from './link-theme';
import formTheme from './form-theme';
import notificationTheme from './notification-theme';

const theme = {
  uiKit: {
    button: buttonTheme,
    link: linkTheme,
    form: formTheme,
    notification: notificationTheme,
  },
};

export default theme;
