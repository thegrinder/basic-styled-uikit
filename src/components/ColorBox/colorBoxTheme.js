const colorBoxTheme = {
  bgColors: {
    transparent: 'transparent',
    white: '#fff',
    lightGray: '#f8f8f8',
    gray: '#666',
    lightPrimary: '#d8eafc',
    primary: '#1e87f0',
    lightSuccess: '#edfbf6',
    success: '#32d296',
    lightWarning: '#fff6ee',
    warning: '#faa05a',
    lightDanger: '#fef4f6',
    danger: '#f0506e',
  },
  borderColors: {
    transparent: 'transparent',
    white: '#fff',
    gray: '#e5e5e5',
    primary: '#1e87f0',
    success: '#32d296',
    warning: '#faa05a',
    danger: '#f0506e',
  },
};

export default colorBoxTheme;
