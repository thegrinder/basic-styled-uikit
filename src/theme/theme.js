import button from '../components/Button/buttonTheme';
import link from '../components/Link/linkTheme';
import form from '../components/Form/formTheme';
import text from '../components/Text/textTheme';
import heading from '../components/Headings/headingTheme';
import colorBox from '../components/ColorBox/colorBoxTheme';

const theme = {
  uiKit: {
    button,
    link,
    form,
    text,
    heading,
    colorBox,
  },
};

export default theme;
