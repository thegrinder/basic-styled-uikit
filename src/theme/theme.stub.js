import { rem } from 'basic-styled-uikit';

const colors = {
  transparent: 'transparent',
  white: '#fff',
  gray50: '#fafafa',
  gray100: '#f8f9fa',
  gray200: '#e9ecef',
  gray300: '#dee2e6',
  gray400: '#ced4da',
  gray500: '#adb5bd',
  gray600: '#868e96',
  gray700: '#495057',
  gray800: '#343a40',
  gray900: '#212529',
  black: '#161616',

  primary50: '#e3f2fd',
  primary100: '#bbdefb',
  primary200: '#90caf9',
  primary300: '#64b5f6',
  primary400: '#42a5f5',
  primary500: '#2196f3',
  primary600: '#1e88e5',
  primary700: '#1976d2',
  primary800: '#1565c0',
  primary900: '#0d47a1',

  success50: '#e8f5e9',
  success100: '#c8e6c9',
  success200: '#a5d6a7',
  success300: '#81c784',
  success400: '#66bb6a',
  success500: '#4caf50',
  success600: '#43a047',
  success700: '#388e3c',
  success800: '#2e7d32',
  success900: '#1b5e20',

  warning50: '#fffde7',
  warning100: '#fff9c4',
  warning200: '#fff59d',
  warning300: '#fff176',
  warning400: '#ffee58',
  warning500: '#ffeb3b',
  warning600: '#fdd835',
  warning700: '#fbc02d',
  warning800: '#f9a825',
  warning900: '#f57f17',

  danger50: '#ffebee',
  danger100: '#ffcdd2',
  danger200: '#ef9a9a',
  danger300: '#e57373',
  danger400: '#ef5350',
  danger500: '#f44336',
  danger600: '#e53935',
  danger700: '#d32f2f',
  danger800: '#c62828',
  danger900: '#b71c1c',
};

const theme = {
  uiKit: {
    colors,
    // button theme
    button: {
      btnTypes: {
        basic: {
          normal: {
            color: colors.gray900,
            backgroundColor: colors.transparent,
            borderColor: colors.gray300,
          },
          hover: {
            color: colors.gray900,
            backgroundColor: colors.transparent,
            borderColor: colors.gray400,
          },
          active: {
            color: colors.gray900,
            backgroundColor: colors.transparent,
            borderColor: colors.gray500,
          },
        },
        default: {
          normal: {
            color: colors.black,
            backgroundColor: colors.gray300,
            borderColor: colors.transparent,
          },
          hover: {
            color: colors.black,
            backgroundColor: colors.gray400,
            borderColor: colors.transparent,
          },
          active: {
            color: colors.black,
            backgroundColor: colors.gray500,
            borderColor: colors.transparent,
          },
        },
        primary: {
          normal: {
            color: colors.white,
            backgroundColor: colors.primary500,
            borderColor: colors.transparent,
          },
          hover: {
            color: colors.white,
            backgroundColor: colors.primary600,
            borderColor: colors.transparent,
          },
          active: {
            color: colors.white,
            backgroundColor: colors.primary700,
            borderColor: colors.transparent,
          },
        },
        danger: {
          normal: {
            color: colors.white,
            backgroundColor: colors.danger500,
            borderColor: colors.transparent,
          },
          hover: {
            color: colors.white,
            backgroundColor: colors.danger600,
            borderColor: colors.transparent,
          },
          active: {
            color: colors.white,
            backgroundColor: colors.danger700,
            borderColor: colors.transparent,
          },
        },
      },
      btnSizes: {
        small: {
          padding: `0 ${rem(15)}`,
          lineHeight: rem(28),
          fontSize: rem(14),
        },
        default: {
          padding: `0 ${rem(30)}`,
          fontSize: rem(16),
          lineHeight: rem(38),
        },
        large: {
          padding: `0 ${rem(40)}`,
          lineHeight: rem(53),
          fontSize: rem(22),
        },
      },
      common: {
        fontWeight: '700',
        letterSpacing: '.4px',
      },
    },
    // form theme
    form: {
      regular: {
        states: {
          valid: {
            normal: {
              backgroundColor: colors.white,
              borderColor: colors.gray300,
              color: colors.gray700,
            },
            active: {
              backgroundColor: colors.white,
              borderColor: colors.primary500,
              color: colors.gray700,
            },
            disabled: {
              backgroundColor: colors.gray100,
              borderColor: colors.gray300,
              color: colors.gray600,
            },
          },
          invalid: {
            backgroundColor: colors.white,
            color: colors.danger500,
            borderColor: colors.danger500,
          },
        },
        common: {},
      },
      inline: {
        states: {
          valid: {
            normal: {
              backgroundColor: colors.transparent,
              borderColor: colors.transparent,
              color: colors.gray700,
            },
            hover: {
              backgroundColor: colors.transparent,
              borderColor: colors.gray300,
              color: colors.gray700,
            },
            active: {
              backgroundColor: colors.transparent,
              borderColor: colors.primary500,
              color: colors.gray700,
            },
            disabled: {
              backgroundColor: colors.gray100,
              borderColor: colors.gray300,
              color: colors.gray600,
            },
          },
          invalid: {
            backgroundColor: colors.transparent,
            color: colors.danger500,
            borderColor: colors.danger500,
          },
        },
        common: {},
      },
      // radio and checkbox theme
      misc: {
        states: {
          valid: {
            normal: {
              borderColor: colors.gray300,
              backgroundColor: colors.transparent,
            },
            active: {
              borderColor: colors.primary500,
              backgroundColor: colors.transparent,
            },
            checked: {
              borderColor: colors.transparent,
              backgroundColor: colors.primary500,
            },
          },
          invalid: {
            borderColor: colors.danger500,
            backgroundColor: colors.transparent,
          },
        },
        common: {},
      },
      // toggle theme
      toggle: {
        normal: {
          backgroundColor: colors.gray400,
        },
        checked: {
          backgroundColor: colors.primary500,
        },
      },
    },
    text: {
      sizings: {
        small: rem(14),
        medium: rem(16),
        large: rem(20),
      },
    },
    heading: {
      sizings: {
        h1: rem(36),
        h2: rem(32),
        h3: rem(28),
        h4: rem(24),
        h5: rem(20),
        h6: rem(18),
      },
    },
  },
};

export default theme;
