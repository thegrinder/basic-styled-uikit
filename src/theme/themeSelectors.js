export { getBoxBgColor, getBoxBorderColor } from '../components/ColorBox/colorBoxHelpers';
export { getHeadingSizing, getHeadingColor } from '../components/Headings/headingHelpers';
export { getTextSizing, getTextColor } from '../components/Text/textHelpers';
