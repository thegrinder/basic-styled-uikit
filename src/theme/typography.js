export const getBaseFontSize = ({ theme }) => theme.uiKit.typography.baseFontSize;

export const getBaseLineHeight = ({ theme }) => theme.uiKit.typography.baseLineHeight;

export const calcLineHeight = () => {};

const typography = {
  baseFontSize: 16,
  baseLineHeight: 1.5,
};

export default typography;
