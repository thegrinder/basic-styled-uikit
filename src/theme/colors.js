import { getColors } from './themeSelectors';

export const getColor = color => ({ theme }) => getColors(theme)[color];

const colors = {
  neutral0: '#fff',
  neutral100: '#f8f9fa',
  neutral400: '#ced4da',
  neutral600: '#868e96',
  neutral1000: '#161616',

  primary400: '#42A5F5',
  primary600: '#1E88E5',
  success400: '#66BB6A',
  success600: '#43A047',
  warning400: '#FFEE58',
  warning600: '#FDD835',
  danger400: '#FF8A65',
  danger600: '#F4511E',
};

export default colors;
