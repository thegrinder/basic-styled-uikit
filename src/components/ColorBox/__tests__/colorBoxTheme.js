import theme from '../../../theme/theme';
import colorBoxTheme, {
  getBoxBgColor,
  getBoxBorderColor,
} from '../colorBoxTheme';

const { bgColors, borderColors } = colorBoxTheme;

describe('colorBoxTheme selectors', () => {
  describe('getBoxBgColor', () => {
    Object.keys(bgColors).forEach((bgColor) => {
      it('should return the correct background color', () => {
        expect(getBoxBgColor(theme, bgColor)).toEqual(bgColors[bgColor]);
      });
    });
  });

  describe('getBoxBgColor', () => {
    Object.keys(borderColors).forEach((borderColor) => {
      it('should return the correct background color', () => {
        expect(getBoxBorderColor(theme, borderColor))
          .toEqual(borderColors[borderColor]);
      });
    });
  });
});
