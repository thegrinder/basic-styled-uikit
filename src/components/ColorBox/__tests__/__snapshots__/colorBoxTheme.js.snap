// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`colorBox should be defined 1`] = `
Object {
  "bgColors": Object {
    "dark": Object {
      "danger": "#ef9a9a",
      "neutral": "#161616",
      "primary": "#90caf9",
      "success": "#a5d6a7",
      "warning": "#fff59d",
    },
    "light": Object {
      "danger": "#f44336",
      "neutral": "#fff",
      "primary": "#2196f3",
      "success": "#4caf50",
      "warning": "#ffeb3b",
    },
  },
  "elevations": Object {
    "dark": Object {
      "1": Object {
        "backgroundColor": "#1b1b1b",
      },
      "2": Object {
        "backgroundColor": "#202020",
      },
    },
    "light": Object {
      "1": Object {
        "boxShadow": "0 3px 1px -2px rgba(0,0,0,.2), 0 2px 2px 0 rgba(0,0,0,.14), 0 1px 5px 0 rgba(0,0,0,.12)",
      },
      "2": Object {
        "boxShadow": "0 2px 4px -1px rgba(0,0,0,.2), 0 4px 5px 0 rgba(0,0,0,.14), 0 1px 10px 0 rgba(0,0,0,.12)",
      },
    },
  },
}
`;
